### PR TITLE
[CICD] Improve Megatron training core unit test coverage

### PR DESCRIPTION
**Pull request by AlexMa616**
# Description

  Improve unit test coverage for Megatron training core modules, focusing on pure Python/control-flow paths that can be tested without changing source behavior or affecting global CI logic. This PR adds and extends unit tests around training argument validation, YAML config conversion, checkp

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] Infra/Build change (changes to CI/CD workflows or build scripts)
- [x] Code refactoring
- [x] Bug fix
- [ ] Documentation change
- [ ] Breaking change

## Changes

  Training coverage: Expanded test_training.py to cover startup timestamp helpers, global state teardown, train/eval/control-flow paths, checkpoint decision paths, data loader/iterator construction, scheduler stepping, skipped-iteration accounting, and validation handling.
  Argument validation coverage: Added validate_args cases for RL generation config resolution, deprecated aliases, MoE argument normalization, tokenizer compatibility paths, skip-train behavior, dtype mapping, and precision guard paths.
  Checkpointing coverage: Added tests for checkpoint format detection, dataloader state save behavior, DTensor conversion, biencoder selective loading, tracker path handling, and checkpoint helper edge cases.
  YAML config coverage: Added tests for YAML-to-core TransformerConfig conversion, activation mapping, MLA config selection, required field validation, and rank-gated argument printing.
  Async/in-process restart coverage: Added tests for persistent async worker setup, async save finalization callbacks, in-process restart wrapper construction, and global state cleanup delegation.
  Stability fixes: Adjusted fragile expectations and fake args to better match actual training control flow, avoid global state leakage, and reduce risk of distributed CI failures.

## Checklist

- [x] I have read and followed the contributing guidelines
- [x] The functionality is complete
- [x] I have added/updated tests that improve CUDA A100 core coverage
- [x] My changes generate no new warnings
- [x] New and existing unit tests are expected to pass in CI
- [x] Local static checks passed where available: py_compile, git diff --check

### CODE REVIEW

File: tests/unit_tests/models/test_mamba_moe_model.py [hunk 1]
```diff
@@ -194,6 +194,7 @@
     "moe_latent_size": None,
     "moe_layer_freq": 1,
     "moe_layer_recompute": False,
+    "moe_mlp_glu_interleave_size": None,
```
Comment (Darryl233) [inline tests/unit_tests/models/test_mamba_moe_model.py:197]: Why changed here ?

Comment (AlexMa616) [reply tests/unit_tests/models/test_mamba_moe_model.py:197]: The newly added model configuration parameters were synchronized to fix the CI configuration interconnection (Config Drift) error in unit tests.

**Review by Darryl233 — APPROVED** [review]
LGTM

**Review by lxd-cumt — APPROVED** [review]
LGTM
